puppetlabs/beaker
# [BKR-623] Reorganize Beaker test runner classes for introduction of minitest runner

![](https://images.rapgenius.com/din5fkgzobu0j1khlnuqjmba9.594x453x1.jpg)

See [BKR-623](https://tickets.puppetlabs.com/browse/BKR-623) for notes on direction. This will likely require a number of refactoring commits to get everything into place.

**TODO**

 - [x] Create `beaker/runner/beaker` + `Beaker::Runner::Beaker` and move  `Beaker::TestSuite` down, leaving a shim class behind
 - [x] Add support for a `--runner` cmdline flag, which defaults to `beaker`
 - [x] use `options[:runner]` to instantiate the appropriate runner from `cli.rb`
 - [x] move `Beaker::TestCase` down into `Beaker::Runner::Beaker`
   - [x] renamed `Beaker::Runner::Beaker` to `Beaker::Runner::Native`, because ruby
 - [x] migrate `TestSuite` tests into runner
 - [x] add a minitest runner `TestSuite` class
 - [x] migrate `TestCase` tests into runner
 - [x] unwind subclassing in `Beaker::TestSuite`
 - [x] remove `, minitest` from `--runner` cmdline docs (not ready for prime time)
 - [x] rename `--runner beaker` to `--runner native`
 - [ ] :eyes: + :ship:
 - [x] structure / outcomes refactoring - see #1023 for ongoing work
   - [ ] open a new Jira ticket for this
   - move "structure" / "outcomes" modules into includes for native `TestCase` class
   - move non-structure/non-outcomes modules into `Beaker::TestCase` class includes
   - clean up `Beaker::DSL`
   - unwind subclassing in `Beaker::TestCase`
   - alias `Beaker::DSL` and `Beaker::TestCase`
 - [ ] add a new Jira ticket for running minitest tests
   - add basic support for minitest runner to execute a series of tests (:warning: this will not work on a beaker suite unless we map structure/outcomes helpers to minitest syntaxes)
   - ensure that `Beaker::TestCase`/`Beaker::DSL` methods are available to minitest runner `TestCase`

/cc @puppetlabs/beaker
